### PR TITLE
Always change language where necessary if path changes

### PR DESCRIPTION
**Pull request by deanc**
A fix for this issue: 
https://github.com/kalinchernev/using-i18n/issues/1

### CODE REVIEW

File: src/layouts/index.js [hunk 1]
```diff
@@ -1,19 +1,24 @@
...
 import withI18next from "../i18n/withI18next";
 
-const Layout = ({ children }) => {
+const Layout = ({ children, location, pageContext: { locale }}) => {
+
+  const { i18n } = useTranslation()
+
+  useEffect(() => {
+    i18n.changeLanguage(locale)
+  }, [location, i18n, locale])
```
Comment (kalinchernev) [inline src/layouts/index.js:15]: What's the idea in mind of adding `i18n` in the dependencies list for the side effect?

Comment (deanc) [reply src/layouts/index.js:15]: Ah, the latest builds of gatsby use eslint which gives this warning:
```
warn ESLintError:
/Users/dc/Projects/finbb-frontpage/src/layouts/index.js
  16:6  warning  React Hook useEffect has missing dependencies: 'i18n' and 'locale'. Either include them or remove the dependency array  react-hooks/exhaustive-deps
```

After a [bit of research](https://github.com/facebook/create-react-app/issues/6903) the recommendation is that you should include all dependencies that are inside the useEffect hook inside that second array, as they can change from the outside too, which in theory should change how they are used inside the hook.

Comment (kalinchernev) [reply src/layouts/index.js:15]: Alright, thanks, that's cool :)
I personally don't like putting whole objects in a dep list, but if the linter say so :P

Comment (deanc) [reply src/layouts/index.js:15]: Sure, but think of it this way. It's used inside the hook, so it is a dependency.

**Comment by kalinchernev** [comment]
It's no worries, for the example here it's good. In actual project some
checks whether its necessary to make the change of language will be easy to
make

**Pull request merged**